features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -512,37 +566,48 @@ nsapi_ip_stack_t AT_CellularNetwork::string_to_stack_type(const char* pdp_type)
     return stack;
 }
 
-nsapi_error_t AT_CellularNetwork::set_registration_urc(bool urc_on)
+nsapi_error_t AT_CellularNetwork::set_registration_urc(RegistrationType type, bool urc_on)
 {
-    for (unsigned int i = 0; i < sizeof(at_reg)/sizeof(at_reg[0]); i++) {
-        if (has_registration(at_reg[i].type)) {
-            _last_reg_type = at_reg[i].type;
-            if (urc_on) {
-                _at.cmd_start(at_reg[i].cmd);
-                _at.write_string("=2", false);
-                _at.cmd_stop();
-            } else {
-                _at.cmd_start(at_reg[i].cmd);
-                _at.write_string("=0", false);
-                _at.cmd_stop();
-            }
-
-            _at.resp_start();
-            _at.resp_stop();
+    int index = (int)type;
+    MBED_ASSERT(index >= 0 && index < C_MAX);
+
+    if (has_registration(type)) {
+        _at.lock();
+        _last_reg_type = type;
+        if (urc_on) {
+            _at.cmd_start(at_reg[type].cmd);
+            _at.write_string("=2", false);
+            _at.cmd_stop();
+        } else {
+            _at.cmd_start(at_reg[type].cmd);
+            _at.write_string("=0", false);
+            _at.cmd_stop();
         }
+
+        _at.resp_start();
+        _at.resp_stop();
+        return _at.unlock_return_error();
+    } else {
+        return NSAPI_ERROR_UNSUPPORTED;
     }
-    return _at.get_last_error();
 }
 
-nsapi_error_t AT_CellularNetwork::set_registration(const char *plmn)
+nsapi_error_t AT_CellularNetwork::get_network_registering_mode(NWRegisteringMode& mode)
 {
     _at.lock();
+    _at.cmd_start("AT+COPS?");
+    _at.cmd_stop();
+    _at.resp_start("+COPS:");
+    int tmpmode = (NWRegisteringMode)_at.read_int();
+    _at.resp_stop();
```

Comment:
Really need here tmpmode variable and casting..?

Reply:
casting yes, tmpmode no...